README.md

```diff
@@ -6,27 +6,106 @@
 solution can assist data lake operators to handle data erasure requests, for
 example, pursuant to the European General Data Protection Regulation (GDPR).
 
-The Amazon S3 Find and Forget solution can be used with Parquet-format data
-stored in Amazon S3 buckets. Your data lake is connected to the solution via an
-AWS Glue table and specifying which columns in the table contain user
-identifiers.
+The solution can be used with Parquet-format data stored in Amazon S3 buckets.
+Your data lake is connected to the solution via AWS Glue tables and by
+specifying which columns in the tables contain user identifiers.
 
 Once configured, you can queue record identifiers that you want the
 corresponding data erased for. You can then run a deletion job to remove the
 data corresponding to the records specified from the objects in the data lake. A
 report log is provided of all the S3 objects modified.
 
+## Installation
+
+The solution is available as an AWS CloudFormation template and should take
+about 20 to 40 minutes to deploy. See the
+[deployment guide](docs/USER_GUIDE.md#deploying-the-solution) to launch the
+deployment with just one click and the
+[cost overview guide](docs/COST_OVERVIEW.md) to learn about its costs.
+
+> Warning: This project is currently in beta release
+>
+> - Actively developed by AWS
+> - Carefully review code and documentation if you intend to use the solution
+>   for production purposes
+> - Get in touch with the team via
+>   [Github Issues](https://github.com/awslabs/amazon-s3-find-and-forget/issues)
+>   for providing feedback and to ask any question. We appreciate your help
+
+## Usage
+
 The solution provides a web user interface, and a REST API to allow you to
 integrate it in your own applications.
 
+![HomePage Screenshot](docs/images/home-screenshot.png)
+
+See the [user guide](docs/USER_GUIDE.md) to learn how to use the solution and
+the [API Specification](docs/api/README.md) to integrate the solution to your
+own applications.
+
+## Architecture
+
+The goal of the solution is to provide a secure, reliable, performant and cost
+effective tool for finding and removing individual records within objects stored
+in S3 buckets. In order to achieve these goals the solution has adopted the
+following design principles:
+
+1. **Security first:** Every component implements least privilege access,
+   encryption is performed at all layers at rest and in transit, authentication
+   is provided out of the box, expiration of logs is easily configurable, and
+   private data is automatically obfuscated or irreversibly deleted as soon as
+   possible when persisting state.
+2. **Built to scale:** The system has been designed and battle tested for
+   performance and high availability with Data Lakes of TB/PB size.
+3. **Cost optimised:**
+   - **Perform work in batches:** Since the time complexity of removing a single
+     vs multiple records in a single object is practically equal and it is
```

Comment:
```suggestion
     vs multiple matches in a single S3 object is practically equal, and it is
```

Reply:
Not sure, do you think that match could be confused with the idea of actually just removing that value and leaving the rest of the row there? Perhaps we can use `rows`?